### PR TITLE
Updates code to terraform 12 and tflinting standards

### PR DESCRIPTION
**Pull request by garybaines**
- Updates code to terraform 12 and tflinting standards
- Adds a minimal terraform version file to the module
- Updates python script to conform to pylint standards
- Adds a CODEOWNERS file for PR approvals set to cloud-engineering

### CODE REVIEW

File: zip_file/versions.tf [hunk 1]
```diff
@@ -0,0 +1,10 @@
+terraform {
+  required_version = "~> 0.12.25"
+
+  required_providers {
+    aws = {
+      #source  = "hashicorp/aws"
```
Comment (michael-dickinson-sainsburys) [inline zip_file/versions.tf]: Is this required?

**Review by michael-dickinson-sainsburys — COMMENTED** [review]
It'd be good to update the readme with details about using the module, including the syntax for using a specific version of it

File: zip_file/versions.tf (hunk 1 above)
Comment (garybaines) [reply zip_file/versions.tf]: Yes and no.  From terraform 13 onwards the commented out line is required to pass tflinting, but it's not compatible with terraform 12, so it's commented out for now.

**Comment by garybaines** [comment]
> It'd be good to update the readme with details about using the module, including the syntax for using a specific version of it

README updated with usage details

File: zip_file/versions.tf (hunk 1 above)
Comment (michael-dickinson-sainsburys) [reply zip_file/versions.tf]: So should we create a version compatible with tf12 without the line, and then a further update for tf13 with the line in?

Comment (garybaines) [reply zip_file/versions.tf]: That is the option.  Go on then, I've pushed it without the comment ;-)

**Review by michael-dickinson-sainsburys — APPROVED** [review]

**Pull request merged**